BUILD

```diff
@@ -0,0 +1,937 @@
+    deps = ["@llvm-project//mlir:PassBaseTdFiles"],
+)
+
+cc_library(
+    name = "stablehlo_passes",
+    srcs = [
+        "stablehlo/transforms/StablehloLegalizeToVhlo.cpp",
+        "stablehlo/transforms/VhloLegalizeToStablehlo.cpp",
+        "stablehlo/transforms/VhloToVersion.cpp",
+    ],
+    hdrs = [
+        "stablehlo/transforms/MapStablehloToVhlo.h",
+        "stablehlo/transforms/Passes.h",
+    ],
+    deps = [
+        ":stablehlo_ops",
+        ":stablehlo_ops_inc_gen",
+        ":stablehlo_pass_inc_gen",
+        ":stablehlo_type_conversion",
+        ":version",
+        ":vhlo_ops",
+        "@llvm-project//llvm:Support",
+        "@llvm-project//mlir:IR",
+        "@llvm-project//mlir:Pass",
+        "@llvm-project//mlir:Support",
+        "@llvm-project//mlir:Transforms",
+    ],
+)
+
+cc_library(
+    name = "stablehlo_type_conversion",
+    srcs = ["stablehlo/transforms/TypeConversion.cpp"],
+    hdrs = ["stablehlo/transforms/TypeConversion.h"],
+    deps = [
+        ":stablehlo_ops",
+        ":vhlo_ops",
+        "@llvm-project//llvm:Support",
+        "@llvm-project//mlir:FuncDialect",
+        "@llvm-project//mlir:FuncTransforms",
+        "@llvm-project//mlir:Transforms",
+    ],
+)
+
+cc_library(
+    name = "reference_element",
+    srcs = [
+        "stablehlo/reference/Element.cpp",
+    ],
+    hdrs = [
+        "stablehlo/reference/Element.h",
+    ],
+    deps = [
+        ":reference_errors",
+        ":reference_types",
+        "@llvm-project//llvm:Support",
+        "@llvm-project//mlir:ComplexDialect",
+        "@llvm-project//mlir:IR",
+        "@llvm-project//mlir:Support",
+    ],
+)
+
+cc_library(
+    name = "reference_errors",
+    hdrs = [
+        "stablehlo/reference/Errors.h",
+    ],
+    deps = [
+        "@llvm-project//llvm:Support",
+    ],
+)
+
+cc_library(
+    name = "reference_interpreter",
+    srcs = [
+        "stablehlo/reference/Interpreter.cpp",
+    ],
+    hdrs = [
+        "stablehlo/reference/Interpreter.h",
+    ],
+    deps = [
+        ":reference_errors",
+        ":reference_ops",
+        ":reference_tensor",
+        ":stablehlo_ops",
+        "@llvm-project//llvm:Support",
+        "@llvm-project//mlir:FuncDialect",
+        "@llvm-project//mlir:IR",
+        "@llvm-project//mlir:Support",
+    ],
+)
+
+cc_library(
+    name = "reference_ops",
+    srcs = [
+        "stablehlo/reference/Ops.cpp",
+    ],
+    hdrs = [
+        "stablehlo/reference/Ops.h",
+    ],
+    deps = [
+        ":reference_element",
+        ":reference_errors",
+        ":reference_tensor",
+        ":reference_types",
+        ":stablehlo_ops",
+        "@llvm-project//llvm:Support",
+        "@llvm-project//mlir:IR",
+        "@llvm-project//mlir:Support",
+    ],
+)
+
+cc_library(
+    name = "reference_tensor",
+    srcs = [
+        "stablehlo/reference/Index.cpp",
+        "stablehlo/reference/Tensor.cpp",
+    ],
+    hdrs = [
+        "stablehlo/reference/Index.h",
```

Comment:
An idiomatic thing to do here would be to create a separate target for Index.h/Index.cpp.

Reply:
I agree, but think it may be best to do reorganizing in a future PR. This looks like it is based on mlir-hlo for now:
https://github.com/tensorflow/mlir-hlo/blob/master/stablehlo/BUILD#L611

Keeping these files as 1:1 as possible for initial submit will probably make integration go smoother. That said I'm not entirely sure what the integration of this file looks like. We can talk offline on that.

Reply:
Sounds good. I was comparing this file with https://github.com/tensorflow/tensorflow/blob/master/third_party/stablehlo/BUILD#L596-L613 and noticed the divergence. Integration into Google3 will help address the discrepancy, and I agree that we don't have to block this PR on that.